.github/workflows/ci.yml

```diff
@@ -0,0 +1,78 @@
+name: CI
+
+on: [push, pull_request]
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        rust:
+          - stable
+          - beta
+          - nightly
```

Comment:
I'm not sure about these. Generally it's good that our project will get built on `beta` and `nightly`, but what if it does not compile on `nightly` or `beta` for some reason and the whole PR is marked as wrong because of that? @agalakhov what do you think?

Reply:
Only build failures on 'stable' are hard errors.  Others are warnings.

Reply:
GH lacks the `allow-failure` feature for specific build jobs but it supports branch protections in `Settings ` to require certain build jobs to pass before merging.

There is a workaround to use `continue-on-error` for  `beta/nightly` build steps to render them green but then you have to check the logs to see if these builds failed.